File: my-folder/0118-pascals-triangle/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        m=[]
        for i in range(1,numRows+1):
            c=1
            t=[]
            for j in range(1,i+1):
                t.append(c)
                c=int(c*(i-j)/j)
            m.append(t)
        return m

File: my-folder/0118-pascals-triangle/solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for(int i=0;i<numRows;i++){
            List<Integer> newList = new ArrayList<>();
            for(int j=0;j<=i;j++){
                if(j==0||j==i){
                    newList.add(1);
                }else{
                    int sum = res.get(i-1).get(j) + res.get(i-1).get(j-1);
                    newList.add(sum);
                }
            }
            res.add(newList);
        }
        
        return res;
    }
}
